source/common/network/utility.cc

```diff
@@ -526,43 +526,84 @@ Api::IoCallUint64Result Utility::writeToSocket(IoHandle& handle, Buffer::RawSlic
   return send_result;
 }
 
+void passPayloadToProcessor(uint64_t bytes_read, Buffer::RawSlice& slice,
+                            Buffer::InstancePtr buffer,
+                            std::shared_ptr<const Address::Instance> peer_addess,
+                            std::shared_ptr<const Address::Instance> local_address,
+                            UdpPacketProcessor& udp_packet_processor, MonotonicTime receive_time) {
+  // Adjust used memory length.
+  slice.len_ = std::min(slice.len_, static_cast<size_t>(bytes_read));
+  buffer->commit(&slice, 1);
+
+  RELEASE_ASSERT(
+      peer_addess != nullptr,
+      fmt::format("Unable to get remote address on the socket bount to local address: {} ",
+                  local_address->asString()));
+
+  // Unix domain sockets are not supported
+  RELEASE_ASSERT(peer_addess->type() == Address::Type::Ip,
+                 fmt::format("Unsupported remote address: {} local address: {}, receive size: "
+                             "{}",
+                             peer_addess->asString(), local_address->asString(), bytes_read));
+  udp_packet_processor.processPacket(std::move(local_address), std::move(peer_addess),
+                                     std::move(buffer), receive_time);
+}
+
 Api::IoCallUint64Result Utility::readFromSocket(IoHandle& handle,
                                                 const Address::Instance& local_address,
                                                 UdpPacketProcessor& udp_packet_processor,
                                                 MonotonicTime receive_time,
                                                 uint32_t* packets_dropped) {
+#if ENVOY_MMSG_MORE
+  const uint32_t num_packets_per_mmsg_call = 16u;
+  const uint32_t num_slices_per_packet = 1u;
+  absl::FixedArray<Buffer::InstancePtr> buffers(num_packets_per_mmsg_call);
+  absl::FixedArray<absl::FixedArray<Buffer::RawSlice>> slices(
```

Comment:
nit: `absl::FixedArray<absl::FixedArray<Buffer::RawSlice>>` is an eyeful, might be a worth some using declarations around the APIs, but up to you.

Reply:
good point! done

Reply:
nit: you can use the using declaration here?